File: selenium_junit5_parallel_with_wdm/src/test/java/org/o2f/pages/automationpractice/components/ResultsComponent.java
package org.o2f.pages.automationpractice.components;

import org.o2f.pages.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;

import java.util.List;

public class ResultsComponent extends BasePage {

    @FindBy(id = "center_column")
    private WebElement resultsContainer;

    @FindBy(className = "lighter")
    private WebElement termSearched;

    @FindBy(className = "product-container")
    private List<WebElement> resultsBoxes;

    public ResultsComponent(WebDriver driver) {
        super(driver);
    }

    public void isLoaded() {
        wait.until(ExpectedConditions.visibilityOf(resultsContainer));
        wait.until(ExpectedConditions.visibilityOf(termSearched));
        wait.until(ExpectedConditions.visibilityOf(resultsBoxes.get(0))); // 1st result loaded
    }

    public String getSearchedTermText() {
        return termSearched.getText().replace("\"", "");
    }

    public int getNumberOfResults() {
        return resultsBoxes.size();
    }
}

File: python-selenium-pure/pages/automation_practice/components/results_component.py
import logging

from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions

from pages.base_page import BasePage


class ResultsComponent(BasePage):

    results_container: By = (By.ID, "center_column")
    term_searched: By = (By.CLASS_NAME, "lighter")
    results_boxes: By = (By.CLASS_NAME, "product-container")

    def __init__(self, driver):
        super().__init__(driver)

    def is_loaded(self):
        self.wait_until_element_displayed(self.results_container)
        self.wait_until_element_displayed(self.term_searched)
        self.wait_until_elements_displayed(self.results_boxes)

    def get_searched_term(self) -> str:
        assert type(self.get_element_text(self.term_searched)) == str
        return str(self.get_element_text(self.term_searched)).replace("\"", "")

    def get_nr_of_results(self) -> int:
        return len(self.wait_until_elements_displayed(self.results_boxes))
